File: design-probes/stateful-index/flows/search/flow.ts
#!/usr/bin/env bun

// DESIGN PROBE ONLY: hypothetical SDK, complete pseudocode behavior.
import { type JsonValue, serve } from "@flow/run";

interface SearchInput {
  readonly query: string;
  readonly limit: number;
}

serve<SearchInput>(async run => {
  const result = await run.effects.call<JsonValue>({
    operationId: "search-index",
    slot: "index",
    method: "search",
    input: run.input as unknown as JsonValue,
  });
  return { outcome: "done", output: result };
});

File: design-probes/cordis-timer/flows/record-firing/flow.py
#!/usr/bin/env python3

# DESIGN PROBE ONLY: hypothetical FLOW Run SDK.
from flow_run import serve


async def record_firing(run):
    event = run.input
    return {
        "outcome": "done",
        "output": {
            "eventId": event["eventId"],
            "timerId": event["data"]["timerId"],
            "payload": event["data"]["payload"],
        },
    }


serve(record_firing)

